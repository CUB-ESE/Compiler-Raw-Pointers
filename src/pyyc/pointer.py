# x = 5 
# print x
x_ptr = ptr(x)
# deref(x_ptr) = 1
# print(deref(x_ptr))
# freep(x_ptr)