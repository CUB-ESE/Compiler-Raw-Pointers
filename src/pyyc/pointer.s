.global main
main:
    pushl %ebp
    movl %esp, %ebp
    subl $8, %esp
    movl $5, -4(%ebp)
    push -4(%ebp)
    call create_ptr
    movl %eax, -8(%ebp)
    push $10
    push -8(%ebp)
    call set_ptr_value
    push -8(%ebp)
    call get_ptr_value
    pushl %eax
    call print_int_nl
    push -8(%ebp)
    call free_ptr
    addl $8, %esp
    movl $0, %eax
    leave
    ret
