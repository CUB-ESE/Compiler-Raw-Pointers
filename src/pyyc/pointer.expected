x = 5 
x_ptr = ptr(x)
deref(x_ptr) = 1
print(deref(x_ptr))
freep(x_ptr)